scripts/globusmonitor/globus_uploader_service.py

```diff
@@ -132,24 +132,48 @@ def connectToPostgres():
     psql_user = os.getenv("POSTGRES_USER", config['postgres']['username'])
     psql_pass = os.getenv("POSTGRES_PASSWORD", config['postgres']['password'])
 
-    try:
-        conn = psycopg2.connect(dbname=psql_db, user=psql_user, password=psql_pass, host=psql_host)
-    except:
-        # Attempt to create database if not found
-        conn = psycopg2.connect(dbname='postgres', host=psql_host)
-        conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
-        curs = conn.cursor()
-        curs.execute('CREATE DATABASE %s;' % psql_db)
-        curs.close()
-        conn.commit()
-        conn.close()
-
-        conn = psycopg2.connect(dbname=psql_db, user=psql_user, password=psql_pass, host=psql_host)
-        initializeDatabase(conn)
+    connected = False
+    total_retry = 0
+    while not connected:
+        try:
+            conn = psycopg2.connect(dbname=psql_db, user=psql_user, password=psql_pass, host=psql_host)
+            connected = True
+        except Exception as e:
+            """Attempt to create database if not found
+            conn = psycopg2.connect(dbname='postgres', host=psql_host)
+            conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+            curs = conn.cursor()
+            curs.execute('CREATE DATABASE %s;' % psql_db)
+            curs.close()
+            conn.commit()
+            conn.close()
+
+            conn = psycopg2.connect(dbname=psql_db, user=psql_user, password=psql_pass, host=psql_host)
+            initializeDatabase(conn)
+            """
+            logger.error("Could not connect to PSQL: %s" % e.message)
+            if total_retry >= 600:
+                logger.error("Exceeded maximum number of retries")
+                raise
+            time.sleep(30)
+            total_retry += 30
 
     logger.info("Connected to Postgres")
     return conn
 
+"""Safer way to call psql_conn.cursor()"""
```

Comment:
The fact that these methods are duplicated across the two files indicates the need for a shared utility module.